core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -147,6 +148,21 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
     unparseFetchUsingLimit(writer, offset, fetch);
   }
 
+  @Override public void unparseByteLiteral(SqlWriter writer,
+      SqlByteLiteral literal, int leftPrec, int rightPrec) {
+    StringBuilder sb = new StringBuilder();
+    sb.append("b'");
+    String hexString = literal.getValue().toString();
+    // Start at 1 and end at length - 1 to remove surrounding single quotes.
```

Comment:
Do we need to worry about whitespace?

Reply:
Good question. I don't think we need to worry about it since the parsing logic won't allow something like that to go through since a white space isn't a hex digit. I can add another comment saying that we are assuming that there are no whitespaces though.